var/spack/repos/builtin/packages/starpu/package.py

```diff
@@ -0,0 +1,148 @@
+    version('1.3.0', sha256='7e06841bddb4b4c7e04e87230cd76b839ef9618d6f4e8ac9b045ec1a80a631c4')
+    version('1.2.8', sha256='b6666dfbc8f248d20df232f27b0edbbb0b426258eac55ec693e45373ba5a75fb')
+    version('1.2.7', sha256='be301a2718d2ca24e125d0c70f98d5bb74525f5908b3d10e63fb980918b5a16f')
+    version('1.2.6', sha256='eb67a7676777b6ed583722aca5a9f63145b842f390ac2f5b3cbc36fe910d964c')
+    version('1.2.5', sha256='1f22eae5c9ee4cae1c8020c5a18f4e36c05c43624bd2ba1f5356a8962d2bdae9')
+    version('1.2.4', sha256='4c99f41a29a5056685a086dc6e299f5bd8c79146f1f0543f1a5882ec5100a9bd')
+    version('1.2.3', sha256='295d39da17ad17752c1cb91e0009fc9b3630bc4ac7db7e2e43433ec9024dc6db')
+    version('1.2.2', sha256='1677990bb9aa8e574634f2407a63aa797313fcb9a0d544b7ac60df90a37e3057')
+    version('1.2.1', sha256='2e0c13f9be7bf7f3c2ce73340ca773d7a0c25bf66a822e44d8fe6407a04f8d77')
+    version('1.2.0', sha256='c0b09a946bcbc680dd169d75856dc759abd98685da777afccc3366998b218a75')
+    version('1.1.6', sha256='c28749b8f0a1586b814cb7d49dffb0f2631dde849591e697c40ef693a8fd54d8')
+
+    version('develop', branch='master')
+    version('master', branch='master')
+    version('git-1.1', branch='starpu-1.1')
+    version('git-1.2', branch='starpu-1.2')
+    version('git-1.3', branch='starpu-1.3')
+
+    variant('shared', default=True, description='Build STARPU as a shared library')
+    variant('fast', default=True, description='Disable runtime assertions')
+    variant('debug', default=False, description='Enable debug symbols')
+    variant('verbose', default=False, description='Enable verbose debugging')
+    variant('fxt', default=False, description='Enable FxT tracing support')
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('cuda', default=False, description='Enable CUDA support')
+    variant('opencl', default=False, description='Enable OpenCL support')
+    variant('openmp', default=True, description='Enable OpenMP support')
+    variant('fortran', default=False, description='Enable Fortran interface and examples')
+    variant('simgrid', default=False, description='Enable SimGrid support')
+    variant('simgridmc', default=False, description='Enable SimGrid model checker support')
+    variant('examples', default=True, description='Enable Examples')
+
+    depends_on("pkg-config", type='build')
+    depends_on("hwloc")
+    depends_on("hwloc+cuda", when='+cuda')
+    depends_on("mpi", when='+mpi~simgrid')
+    depends_on("cuda", when='+cuda~simgrid')
+    depends_on("fxt", when='+fxt')
+    depends_on("simgrid", when='+simgrid')
+    depends_on("simgrid+smpi", when='+simgrid+mpi')
+    depends_on("simgrid+mc", when='+simgridmc')
+
+    conflicts('+shared', when='+mpi+simgrid', msg="Simgrid MPI require to disable shared library")
+
+    def autoreconf(self, spec, prefix):
+        if not os.path.isfile("./configure"):
+            autogen = Executable("./autogen.sh")
```

Comment:
As above: This probably requires `autoconf`, `libtool`, and maybe other packages that need to be added as dependencies.

Does the existence of `configure` correlate with a version? For example, released versions always have a `configure` file, while the git branches don't? In this case, the `autoconf` can be a conditional build dependency. `libtool` might still be required all the time.